front/lib/api/tables.ts

```diff
@@ -311,61 +319,65 @@ export async function rowsFromCsv(
     }
 
     const record = anyRecord as string[];
+    records.push(record);
 
-    if (!header) {
-      header = [];
-      const firstRecordCells = record.map((h) => h.trim().toLocaleLowerCase());
-      const firstEmptyCellIndex = firstRecordCells.indexOf("");
-      if (firstEmptyCellIndex !== -1) {
-        // Ensure there are no non-empty cells after the first empty cell.
-        if (firstRecordCells.slice(firstEmptyCellIndex).some((c) => c !== "")) {
-          return new Err({
-            type: "invalid_header",
-            message: `Invalid header: found non-empty cell after empty cell.`,
-          });
-        }
-      }
-      header = firstRecordCells.slice(
-        0,
-        firstEmptyCellIndex === -1 ? undefined : firstEmptyCellIndex
-      );
-      const headerSet = new Set<string>();
-      for (const h of header) {
-        if (!isSlugified(h)) {
-          return new Err({
-            type: "invalid_header",
-            message: `Header '${h}' is not a valid slug. A header should only contain lowercase letters, numbers, or underscores.`,
-          });
-        }
-
-        if (headerSet.has(h)) {
-          return new Err({
-            type: "duplicate_header",
-            message: `Duplicate header: ${h}.`,
-          });
-        }
-        headerSet.add(h);
-      }
-      continue;
+    if (records.length === 10) {
+      break;
     }
+  }
+  headParser.destroy();
 
-    for (const [i, h] of header.entries()) {
-      const col = record[i];
-      if (col === undefined) {
-        return new Err({
-          type: "invalid_record_length",
-          message: `Invalid record length at row ${i} (expected ${header.length} columns, got ${record.length})).`,
-        });
-      }
-      if (!valuesByCol[h]) {
-        valuesByCol[h] = [col];
-      } else {
-        (valuesByCol[h] as string[]).push(col);
+  const action = DustProdActionRegistry["table-header-parser"];
+
+  const model = getLargeWhitelistedModel(auth.getNonNullableWorkspace());
```

Comment:
expensive. too expensive?

Reply:
I actually replaced with getSmallWhitelistedModel , clearly not needed to use a large model. But still expensive, indeed.

Reply:
Do we have approximate numbers?

Reply:
In terms of time, it takes between 1 and 2 sec, using app on production dust.tt from my local. Some runs are available here : https://dust.tt/w/78bda07b39/a/OJSd3KpoDU/runs?wIdTarget=0ec9852c2f

Reply:
We're inherently async here so 2s is fine. My question was more about anticipated $ cost 🙏

Reply:
Yes, exactly, that was mainly my point and reason for asking :-) 
The number of input token should be quite small, only the prompt and the 10 first lines of the csv. And output is very small (but can be bigger if we have tables with a lot of of columns, we should add a limit here)

Reply:
Didn't ask the original question?? :D

Let's compute a rough daily cost estimate. Maybe worth digging in how much caching would save (we have it built in Dust apps)

Reply:
I was asking in slack :-) I've checked the total number of table upserts per day, we are at about 100k average : 

https://app.datadoghq.eu/logs?query=%22%5BUpsertQueue%5D%20Successful%20table%20upsert%22%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host,service&fn=%5B%5B%22ewma_20%22%5D%5D&fromUser=true&messageDisplay=inline&step=%7B%22type%22:%22day%22%7D&storage=hot&stream_sort=desc&viz=timeseries&from_ts=1724224062258&to_ts=1725520062258&live=true

In terms of cost, a run is about 0.0001$ : https://dust.tt/w/0ec9852c2f/assistant/bCF4BRVF1J  ( I've got a small amount of runs for now, mainly based on our google drive spreadsheets indexation ) - so if we enable on all workspaces, it would be something like 10$ per day . 

Considering we may reupsert the same tables (or same headers) multiple times, caching will probably help, but I don't have the exact figures now - we'll need to enable this on a single workspace to check how it behaves.